camkes/templates/component.template.h

```diff
@@ -213,6 +213,7 @@ const char *get_instance_name(void);
     #define /*? d.name ?*/_release() COMPILER_MEMORY_RELEASE()
     #define /*? d.name ?*/_acquire() COMPILER_MEMORY_ACQUIRE()
 
+    #define /*? d.name ?*/_size /*? macros.dataport_size(d.type) ?*/
```

Comment:
This seems fine to me.  My only question is if it's better to define as a function macro so that it's easier to implement as a function call in the future if it becomes necessary.

Reply:
So do I understand correctly that your suggestion would be something like:
```suggestion
    #define /*? d.name ?*/_get_size() /*? macros.dataport_size(d.type) ?*/
```
And then adapt the documentation and place the additional function not in the exposed variables but the section about the available functions at runtime?

Reply:
Maybe provide both? The define make it accessible to the compiler, that has certain advantages for static code. This is not possible when we have a function. Didn't we have a similar discussion on anotherCAmkES PR some time ago?

Reply:
> So do I understand correctly that your suggestion would be something like:
> 
> And then adapt the documentation and place the additional function not in the exposed variables but the section about the available functions at runtime?

Yes.  Although in the documentation maybe should just say it is a macro as it is neither a function or a variable currently.

> Maybe provide both? The define make it accessible to the compiler, that has certain advantages for static code.

It's should still stay as a macro for now.

Reply:
I updated the commit with a new suggestion, @kent-mcleod could you please take another look if this is more aligned with your thoughts?

Reply:
The way I see it is, that for practical reason (ie. define static arrays in C) we desperately need: 
```
#define /*? d.name ?*/_size /*? macros.dataport_size(d.type) ?*/
```
Now optionally, having the additional function 
```
static inline size_t /*? d.name ?*/_get_size(void) { 
    return /*? macros.dataport_size(d.type) ?*/;
}
```
would be nice, so people can write cleaner code and don't use the macro when the don't have to. Documentation for the macro could say that this also, the macro only exists, so fixed size arrays can be define and there is no need to use malloc(). And that this anyone day no longer work when CAmkES becomes more dynamic. But it this will not really happen in the next few years, then nobody know how the dynamic thing might look like and there might be code that can deal with this anyway. So it might be more likely there is a new "dataport" type then for this that does not break compatibility.

Reply:
I was only suggesting doing the following: 
`#define /*? d.name ?*/_size() /*? macros.dataport_size(d.type) ?*/`

Reply:
I think that mimicking function by a macro is a bad choice, because it does not really align with C semantics. You cannot do a `void *fn_ptr = &dataport_foo_size` there. Also, `uint8_t some_array[dataport_foo_size]` does not work, but you have to use `uint8_t some_array[dataport_foo_size()]`, But that is odd, because you can't use function for array sizes actually. And if `dataport_foo_size()` becomes a real function one day, the array declaration will no longer work.
That is why I think there is no other choice but define the macro and the function.

Reply:
Sure.